utilities/driver-checksum.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+function process_gcp_driver() {
+    checksum="${1}"
+    file_in_bucket="${2}"
+    retval=0
+
+    temp_dir="$(mktemp -d)"
+    driver="${temp_dir}/driver"
+
+    echo "Downloading ${file_in_bucket} to ${driver}"
+
+    gsutil cp "${file_in_bucket}" "${driver}"
+
+    if ! process_file_driver "${checksum}" "${driver}"; then
+        retval=1
+    fi
+
+    rm -rf "${temp_dir}"
+    return $retval
+}
+
+function process_file_driver() {
+    retval=0
+
+    temp_file=$(mktemp)
+    echo "${1} ${2}" > "${temp_file}"
+
+    if ! sha256sum -c "${temp_file}"; then
+        echo >&2 "Checksum validation failed"
+        retval=1
+    fi
+
+    rm -f "${temp_file}"
+    return $retval
+}
+
+SOURCE=${1}
```

Comment:
```suggestion
SOURCE=${1} # Valid options are "gcp" or "file"
```

Reply:
I think SOURCE_TYPE is clearer than SOURCE. This will need to be changed elsewhere as well.

Reply:
I tend to think of "source" as "the place the driver we are testing comes from", so IMO source is actually more accurate than source_type which would imply there is a source somewhere else in the code, WDYT?